src/neomaril_codex/model.py

```diff
@@ -228,33 +236,43 @@ def restart_model(self, *, wait_for_ready: bool = True):
         -------
         >>> model.restart_model()
         """
-        if (self.operation == "sync") and (self.status == "Deployed"):
-            url = f"{self.base_url}/model/sync/restart/{self.group}/{self.model_id}"
-            response = requests.get(
-                url,
-                headers={
-                    "Authorization": "Bearer "
-                    + refresh_token(*self.credentials, self.base_url)
-                },
-            )
-            if response.status_code == 200:
-                logger.info("Model is restarting")
-                self.status = self.__get_status()["Status"]
-                if wait_for_ready:
-                    print("Waiting for deploy to be ready.", end="")
-                    while self.status == "Building":
-                        sleep(30)
-                        self.status = self.__get_status()["Status"]
-                        print(".", end="", flush=True)
-            elif response.status_code == 401:
-                logger.error(response.text)
-                raise AuthenticationError("Login not authorized")
-            elif response.status_code >= 500:
-                logger.error(response.text)
-                raise ServerError("Server Error")
-            else:
-                logger.error(response.text)
-                raise ModelError("Could not restart the model")
+        if self.status not in [
```

Comment:
shouldn't this check be on the backend side?

Reply:
Good point. I was thinking about it and I agree that the backend should validate this, but would double validation be good? Also, imagine I remove all validation points in the code. There's no reason to keep the status attribute, right? This attribute is useful only for validations (at least I don't see any other reason for it). What do you think?